BlobStorage/Advanced.cs

```diff
@@ -1584,80 +1332,67 @@ private static async Task CopyBlockBlobAsync(CloudBlobContainer container)
+        private static async Task CopyLargeBlockBlobAsync(BlobContainerClient container, int sizeInMb)
         {
             // Create an array of random bytes, of the specified size.
             byte[] bytes = new byte[sizeInMb * 1024 * 1024];
             Random rng = new Random();
             rng.NextBytes(bytes);
 
             // Get a reference to a new block blob.
-            CloudBlockBlob sourceBlob = container.GetBlockBlobReference("LargeSourceBlob");
+            BlobClient sourceBlob = container.GetBlobClient("LargeSourceBlob");
 
             // Get a reference to the destination blob (in this case, a new blob).
-            CloudBlockBlob destBlob = container.GetBlockBlobReference("copy of " + sourceBlob.Name);
+            BlobClient destBlob = container.GetBlobClient("copy of " + sourceBlob.Name);
 
-            MemoryStream msWrite = null;
-            string copyId = null;
             string leaseId = null;
 
             try
             {
                 // Create a new block blob comprised of random bytes to use as the source of the copy operation.
-                msWrite = new MemoryStream(bytes);
-                msWrite.Position = 0;
-                using (msWrite)
-                {
-                    await sourceBlob.UploadFromStreamAsync(msWrite);
-                }
+                await sourceBlob.UploadAsync(BinaryData.FromString(bytes.ToString()));
 
                 // Lease the source blob for the copy operation to prevent another client from modifying it.
                 // Specifying null for the lease interval creates an infinite lease.
-                leaseId = await sourceBlob.AcquireLeaseAsync(null);
+                var leaseClient = sourceBlob.GetBlobLeaseClient();
+                var blobLease = await leaseClient.AcquireAsync(TimeSpan.FromSeconds(15));
+                leaseId = blobLease.Value.LeaseId;
 
                 // Get the ID of the copy operation.
-                copyId = await destBlob.StartCopyAsync(sourceBlob);
-
-                // Fetch the destination blob's properties before checking the copy state.
-                await destBlob.FetchAttributesAsync();
+                CopyFromUriOperation copyFromUriOperation = await destBlob.StartCopyFromUriAsync(sourceBlob.Uri);
+                string copyId = copyFromUriOperation.Id;
 
                 // Sleep for 1 second. In a real-world application, this would most likely be a longer interval.
                 System.Threading.Thread.Sleep(1000);
 
                 // Check the copy status. If it is still pending, abort the copy operation.
-                if (destBlob.CopyState.Status == CopyStatus.Pending)
+                if ((await destBlob.GetPropertiesAsync()).Value.CopyStatus == CopyStatus.Pending)
                 {
-                    await destBlob.AbortCopyAsync(copyId);
+                    await destBlob.AbortCopyFromUriAsync(copyId);
```

Comment:
The whole point of a `CopyFromUriOperation` is that it can do the polling for you. We should be using this object's methods instead of checking blob properties for the copy status.